### ISSUE TITLE
You shouldn't be able to instantiate `Never` or other special forms

### ISSUE THREAD

**Issue by not-my-profile** [microsoft/pyright#4373]
```python
from typing import Never, Self, Unpack, Required, NotRequired, LiteralString

Never()
Self()
Unpack()
Required()
NotRequired()
LiteralString()
```

Pyright 1.1.285 doesn't complain about any of these, even though they result in runtime `TypeError`s, whereas mypy correctly reports `error: "_SpecialForm" not callable  [operator]`.

Sidenote: I think the output of `reveal_type(Never())` is quite funny:

> information: Type of "Never()" is "Never"

but at least pyright disallows `Never()` to be passed as `Never`, even though the error message is quite confusing:

> Type "Never" cannot be assigned to type "Never"

**Comment by erictraut** [comment]
I would find it surprising if anyone ever attempted to instantiate these special forms. Adding checks for things that will likely never be attempted just serves to slow down type checking and reduce maintainability of the type checking code, so I'm inclined to reject this feature request for now. We might reconsider in the future based on additional signals from users.
